### DISCUSSION TITLE
Every new deploy uses different NEXT_PUBLIC_ env var

### DISCUSSION THREAD

**Discussion by Yuripetusko** [vercel/community#1534] [Help] [answered]
We have the same  env variable set in vercel for different preview branches with different values. For the past couple of weeks (around the time we updated to next13), when we deploy a branch it seems like a build alternates between what env. variable to use. Before that, we never had this issue and Preview env.  var with branch specified was always used in place of generic Preview env. var or Development env. var.

I am not sure where to look exactly, and it's hard to reproduce this consistently. Maybe someone here has some insight on why this might be happening?

**Comment by Yuripetusko** [comment]
Just happened again and it looks like it's "caching" or using the env. var values of previous Preview build in the same project ignoring env var set for this branch 

Deploy 1: `NEXT_PUBLIC_FOO` is set to `TRUE` on branch `staging` (`Preview on staging`) in Vercel. Deployed URL is fine
Deploy 2: `NEXT_PUBLIC_FOO` is set to `FALSE` on branch `development` (`Preview on development`) in Vercel. Deployed URL still using `TRUE` from previous deploy

Then if I click Redeploy on vercel, the `development` branch deploy will get a correct env var value

**Comment by amyegan** [comment]
Thanks for reaching out, Yuri. I wasn't able to reproduce with problem in my test project, but you already mentioned it's not a consistent problem. 😞 

Are you able to share your repo or [MRE](https://vercel.com/support/articles/creating-a-minimal-reproducible-example)? I will pass this info along to the deployment engineers either way, but testing with your code will probably increase their chances of seeing the problem occur

**Comment by Yuripetusko** [comment, marked as answer]
Looks like it is solved after we added these to our nx.json to make sure nx cache is dropped if these env vars are different.

```
"targetDefaults": {
    "build": {
      "dependsOn": [
        "^build"
      ],
      "inputs": [
        "production",
        "^production",
        {
          "env": "NEXT_PUBLIC_ENVIRONMENT" // we use this a lot 
        },
        {
          "env": "VERCEL_ENV"
        }
      ]
    },
```
